core/src/main/scalajvm/sttp/client4/internal/httpclient/WebSocketImpl.scala

```diff
@@ -55,29 +52,61 @@ private[client4] class WebSocketImpl[F[_]](
           // making close sequentially idempotent
           if (wasOpen) fromCompletableFuture(ws.sendClose(statusCode, reasonText)) else ().unit
       }
-    }))
+    }
 
   override lazy val upgradeHeaders: Headers = Headers(Nil)
 
   override def isOpen(): F[Boolean] = monad.eval(_isOpen.get())
 
-  override implicit def monad: MonadError[F] = _monad
-
-  private def fromCompletableFuture(cf: CompletableFuture[JWebSocket]): F[Unit] =
-    _monad.async { cb =>
-      cf.whenComplete(new BiConsumer[JWebSocket, Throwable] {
-        override def accept(t: JWebSocket, error: Throwable): Unit =
-          if (error != null) {
-            cb(Left(error))
-          } else {
-            cb(Right(()))
-          }
-      })
-      Canceler { () =>
-        cf.cancel(true)
-        ()
-      }
+  protected[client4] def fromCompletableFuture(cf: CompletableFuture[JWebSocket]): F[Unit]
+
+}
+
+private[client4] object WebSocketImpl {
+  def sync[F[_]](
```

Comment:
I think we still need a sequencer for the sync version - a sync websocket can still be used from multiple threads. I think using a semaphore as in `FutureSequencer` should work here (but in a sync version, of course)